test/e2e-internal-encryption-tests.sh

```diff
@@ -14,6 +14,76 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-echo "TODO(KauzClay): Implement Me!"
+source $(dirname $0)/e2e-common.sh
+
+function setup_internal_encryption_env_variables() {
+  export TLS_TEST_NAMESPACE="tls"
+
+  local INGRESS_NAMESPACE=${GATEWAY_NAMESPACE_OVERRIDE}
+  if [[ -z "${GATEWAY_NAMESPACE_OVERRIDE}" ]]; then
+    INGRESS_NAMESPACE="istio-system"
+  fi
+  local INGRESS_SERVICE=${GATEWAY_OVERRIDE}
+  if [[ -z "${GATEWAY_OVERRIDE}" ]]; then
+    INGRESS_SERVICE="istio-ingressgateway"
+  fi
+  local IP=$(kubectl get svc -n ${INGRESS_NAMESPACE} ${INGRESS_SERVICE} -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+  export INTERNAL_ENCRYPTION_TEST_INGRESS_IP=${IP}
+}
+
+function setup_internal_encryption() {
+  toggle_feature dataplane-trust enabled config-network
+
+  sleep 5
+
+  # with the current implementation, Activator is always in the request path, and needs to be restarted after configuring dataplane-trust
+  kubectl -n ${SYSTEM_NAMESPACE} delete pod -l app=activator
+}
+
+function cleanup_internal_encryption() {
+  toggle_feature dataplane-trust disabled config-network
+
+  sleep 5
+
+  # with the current implementation, Activator is always in the request path, and needs to be restarted after configuring dataplane-trust
+  kubectl -n ${SYSTEM_NAMESPACE} delete pod -l app=activator
+}
+
+# Script entry point.
+initialize "$@" --skip-istio-addon --min-nodes=4 --max-nodes=4 --enable-ha --cluster-version=1.25
+
+# Run the tests
+header "Running tests"
+
+failed=0
+
+# Currently only Contour and Kourier implement the alpha features.
+alpha=""
+if [[ "${INGRESS_CLASS}" == "contour.ingress.networking.knative.dev" || "${INGRESS_CLASS}" == "kourier.ingress.networking.knative.dev" ]]; then
+  alpha="--enable-alpha"
+fi
+
+INTERNAL_ENCRYPTION_TEST_OPTIONS="${INTERNAL_ENCRYPTION_TEST_OPTIONS:-${alpha} --enable-beta}"
+
+# Auto TLS E2E tests mutate the cluster and must be ran separately
+# because they need auto-tls and cert-manager specific configurations
+subheader "Setup internal encryption"
+setup_internal_encryption
+add_trap "cleanup_internal_encryption" EXIT SIGKILL SIGTERM SIGQUIT
```

Comment:
Eeep... any way this could be less shell, or have a little block explaining why it's shell?

Reply:
(Maybe "see the README" is sufficient...)

Reply:
ah yeah, this part was one I wasn't so sure on. I was under the impression you needed a script like this if you needed to change a config on the test cluster (like setting dataplane-trust), so I just tried to base things on the [auto-tls tests](https://github.com/knative/serving/blob/main/test/e2e-auto-tls-tests.sh). 

But I'll be honest, I don't completely understand how that script ties into the rest of the tooling so that the tests actually run.

Reply:
okay, I got a little clarification, and I think I can move the core behavior here into the `e2e-tests.sh` file, and get rid of the file itself.